pkg/execution/executor/executor.go

```diff
@@ -2439,9 +2439,13 @@ func (e *executor) extractTraceCtx(ctx context.Context, id state.Identifier, ite
 	return ctx
 }
 
-// AppendAndScheduleBatch appends a new batch item. If a new batch is created, it will be scheduled to run
+func (e *executor) AppendAndScheduleBatch(ctx context.Context, fn inngest.Function, bi batch.BatchItem) error {
```

Comment:
why not do variadic variables for opts instead? like
```
AppendAndScheduleBatch(ctx context.Context, fn inngest.Function, bi batch.BatchItem, opts ...BatchOpt)
```

This way, you can call the function with or without the 4th argument.

Reply:
I started down that path, but:

If the options are a trivial struct (as I've written in this PR), we need to handle the case where someone passes multiple options. (In particular, what if someone passes conflicting options?) We could return an error if given >1 options structure, or we could say that the first or last one takes priority, but overall it makes for a confusing API.

If the options are functions (following e.g. [the functional options pattern](https://davidbacisin.com/writing/golang-options-pattern)), then we still need to add some sort of structure for those functions to modify. It's not clear to me what that structure should be, particularly since this option is used to build different types of requests.

Ultimately, I couldn't come up with a good, usable API using varargs, and I think it's clear how to use this API correctly. However, if you feel strongly that we don't want to add new methods here, I'm willing to switch back to varargs and make the API as ergonomic as I can.